substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrManager.java

```diff
@@ -217,6 +204,44 @@ private static Long parseDuration(Map<JfrStartArgument, String> args, JfrStartAr
         return null;
     }
 
+    private static Long parseMaxSize(Map<JfrStartArgument, String> args, JfrStartArgument key) {
+        final String value = args.get(key);
+        if (value != null) {
+            try {
+                int idx = indexOfFirstNonDigitCharacter(value);
+                long number;
+                try {
+                    number = Long.parseLong(value.substring(0, idx));
+                } catch (NumberFormatException e) {
+                    throw new IllegalArgumentException("Expected a number.");
+                }
+
+                // Missing unit, number is plain bytes
+                if (idx == value.length()) {
+                    return number;
+                }
+
+                final char unit = value.substring(idx).charAt(0);
+                switch (unit) {
+                    case 'k':
+                    case 'K':
+                        return number * 1024;
+                    case 'm':
+                    case 'M':
+                        return number * 1024 * 1024;
+                    case 'g':
+                    case 'G':
+                        return number * 1024 * 1024 * 1024;
+                    default:
+                        return number;
```

Comment:
Could a comment be added here to explain the default case?

Reply:
Rest LGTM!

Reply:
> Could a comment be added here to explain the default case?

Comment added. Squashed into the previous commit :+1: